airbyte-webapp/src/core/form/schemaToYup.ts

```diff
@@ -0,0 +1,208 @@
+import { JSONSchema7 } from "json-schema";
+import * as yup from "yup";
+
+import { FormBlock, FormGroupItem, FormObjectArrayItem, FormConditionItem } from "core/form/types";
+import { isDefined } from "utils/common";
+
+import { FormBuildError } from "./FormBuildError";
+
+/**
+ * Returns yup.schema for validation
+ *
+ * This method builds yup schema based on jsonSchema ${@link JSONSchema7} and the derived ${@link FormBlock}.
+ * Every property is walked through recursively in case it is condition | object | array.
+ *
+ * @param jsonSchema
+ * @param formField The corresponding {@link FormBlock} to the given schema
+ * @param parentSchema used in recursive schema building as required fields can be described in parentSchema
+ * @param propertyKey used in recursive schema building for building path for uiConfig
+ * @param propertyPath constructs path of property
+ */
+export const buildYupFormForJsonSchema = (
+  jsonSchema: JSONSchema7,
+  formField: FormBlock,
+  parentSchema?: JSONSchema7,
+  propertyKey?: string,
+  propertyPath: string | undefined = propertyKey
+): yup.AnySchema => {
+  let schema:
+    | yup.NumberSchema
+    | yup.StringSchema
+    | yup.AnyObjectSchema
+    | yup.ArraySchema<yup.AnySchema>
+    | yup.BooleanSchema
+    | null = null;
+
+  if (jsonSchema.oneOf && propertyPath) {
+    const conditionFormField = formField as FormConditionItem;
+    // for all keys in all of the possible objects from the oneOf, collect the sub yup-schema in a map.
+    // the keys of the map are the keys of the property, the value is an array of the selection const value
+    // for the condition plus the sub schema for that property in that condition.
+    // As not all keys will show up in every condition, there can be a different number of possible sub schemas
+    // per key; at least one and at max the number of conditions (if a key is part of every oneOf)
+    const flattenedKeys: Map<string, Array<[unknown, yup.AnySchema]>> = new Map();
```

Comment:
Why is the value type of the map `Array<[unknown, yup.AnySchema]>`? Shouldn't it be `Array<[string, yup.AnySchema]>`, since all properties are strings?

Reply:
The keys of the map correspond with the property keys in each of the possible `oneOf`s - those need to be strings. But the array of `Array<[unknown, yup.AnySchema]>` is a list of mappings from the value of the const property to the schema it implies - const properties can be anything, they are not limited to strings (in practice they are probably always strings, but numbers could be the case as well).

This is where the type for these comes from:
<img width="1052" alt="Screenshot 2022-12-15 at 12 13 15" src="https://user-images.githubusercontent.com/1508364/207845267-4633bee8-342d-49af-8b71-925f815a9708.png">

`JSONSchema7Type` is pretty much everything you want:
<img width="677" alt="Screenshot 2022-12-15 at 12 14 15" src="https://user-images.githubusercontent.com/1508364/207845390-f9da3738-6727-4b69-84a0-a4197d0c9438.png">

Reply:
Would it work to make the type of `FormConditionItem.selectionConstValues` be something like `Array<JSONSchema7Type | undefined>` then?

Just asking because it feels like we should try to avoid `unknown` where we can and this feels like a place where we may be able to eliminate it

Reply:
I'm not too worried about using `unknown` as it's a clear marker that you shouldn't assume anything about this value -  it basically allows you to do nothing with it except for passing it around and checking whether it's the same value. 

Switching to `JSONSchema7Type` is closer to the truth though - updated the code (we can even drop the "undefined" as this would mean in one condition no `const` is set which is something we throw on).